src/metabase/lib/filter.cljc

```diff
@@ -59,45 +63,162 @@
+
+      [:= _ (a :guard numeric?) & args]
+      (i18n/tru "{0} is equal to {1} selections" (->display-name a) (count args))
+
+      [:!= _ (a :guard numeric?) & args]
+      (i18n/tru "{0} is not equal to {1} selections" (->display-name a) (count args))
+
+      [:!= _ (a :guard temporal?) & args]
+      (i18n/tru "{0} excludes {1} {2} selections" (->unbucketed-display-name a) (count args) (->bucket-name a))
+
+      [:= _ a & args]
+      (i18n/tru "{0} is {1} selections" (->display-name a) (count args))
+
+      [:!= _ a & args]
+      (i18n/tru "{0} is not {1} selections" (->display-name a) (count args)))))
 
 (defmethod lib.metadata.calculation/display-name-method ::binary
-  [query stage-number [tag _opts x y] style]
-  (let [x (lib.metadata.calculation/display-name query stage-number x style)
-        y (lib.metadata.calculation/display-name query stage-number y style)]
-    (clojure.core/case tag
-      :<                (i18n/tru "{0} is less than {1}"                x y)
-      :<=               (i18n/tru "{0} is less than or equal to {1}"    x y)
-      :>                (i18n/tru "{0} is greater than {1}"             x y)
-      :>=               (i18n/tru "{0} is greater than or equal to {1}" x y)
-      :starts-with      (i18n/tru "{0} starts with {1}"                 x y)
-      :ends-with        (i18n/tru "{0} ends with {1}"                   x y)
-      :contains         (i18n/tru "{0} contains {1}"                    x y)
-      :does-not-contain (i18n/tru "{0} does not contain {1}"            x y))))
+  [query stage-number expr style]
+  (let [->display-name #(lib.metadata.calculation/display-name query stage-number % style)
+        ->temporal-name #(shared.ut/format-unit % nil)
+        temporal? #(lib.util/original-isa? % :type/Temporal)]
+    (mbql.u/match-one expr
+      [:< _ (x :guard temporal?) (y :guard string?)]
+      (i18n/tru "{0} is before {1}"                   (->display-name x) (->temporal-name y))
+
+      [:< _ x y]
+      (i18n/tru "{0} is less than {1}"                (->display-name x) (->display-name y))
+
+      [:<= _ x y]
+      (i18n/tru "{0} is less than or equal to {1}"    (->display-name x) (->display-name y))
+
+      [:> _ (x :guard temporal?) (y :guard string?)]
+      (i18n/tru "{0} is after {1}"                    (->display-name x) (->temporal-name y))
+
+      [:> _ x y]
+      (i18n/tru "{0} is greater than {1}"             (->display-name x) (->display-name y))
+
+      [:>= _ x y]
+      (i18n/tru "{0} is greater than or equal to {1}" (->display-name x) (->display-name y))
+
+      [:starts-with _ x (y :guard string?)]
+      (i18n/tru "{0} starts with {1}"                 (->display-name x) y)
+
+      [:starts-with _ x y]
+      (i18n/tru "{0} starts with {1}"                 (->display-name x) (->display-name y))
+
+      [:ends-with _ x (y :guard string?)]
+      (i18n/tru "{0} ends with {1}"                   (->display-name x) y)
```

Comment:
Why not define `->display-name` for strings as identity?

Reply:
String as identity would affect custom expressions that would have a larger chance to be ambiguous between columns and strings. I didn't want to expand the scope more than we needed.